### PR TITLE
[ASCII-984] Fix dead URLs in markdown files

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Fix dead or invalid links in markdown files. 

### Motivation

Documentation shouldn't have dead links.

### Additional Notes

I used https://github.com/tcort/markdown-link-check to detect dead links.
I'll add a [Github workflow](https://github.com/gaurav-nelson/github-action-markdown-link-check) to be able to detect this kind this kind of issue (soon-ish).

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: docs/dev/agent_dev_env.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ sudo apt-get install python3.9-dev
 On Windows, install Python 2.7 and/or 3.9 via the [official installer](https://www.python.org/downloads/).
 
 #### Additional Windows Tools
-You will also need the Visual Studio for [Visual Studio for Python installer](http://aka.ms/vcpython27)
+You will also need the [Visual Studio for Python installer](https://visualstudio.microsoft.com/fr/vs/features/python/).
```
Comment (pgimalac) [inline docs/dev/agent_dev_env.md:34]: This is a best guess

File: docs/dev/checks/README.md [hunk 2]
```diff
@@ -127,7 +127,7 @@ checks code.
 
 [custom-checks]: https://docs.datadoghq.com/developers/write_agent_check/?tab=agentv6
 [collector]: /pkg/collector
-[datadog_checks_base]: https://datadog-checks-base.readthedocs.io/en/latest/
+[datadog_checks_base]: https://datadoghq.dev/integrations-core/base/about/
```
Comment (pgimalac) [inline docs/dev/checks/README.md:130]: This is a best guess

File: docs/proposals/TEMPLATE.md
Comment (pgimalac) [inline docs/proposals/TEMPLATE.md:1]: Doesn't seem to have really ever been used

File: pkg/collector/README.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ a check are all operations implemented in this package or one of the subpackages
 For further details, please refer to the specific READMEs:
 
 * [check](check/README.md)
-* [corechecks](corechecks/README.md)
-* [py](py/README.md)
-* [runner](runner/README.md)
+* [corechecks](corechecks)
+* [python](python)
+* [runner](runner)
```
Comment (pgimalac) [inline pkg/collector/README.md:12]: No README file in those directories, not sure the items should just be removed ?

Comment (vickenty) [reply pkg/collector/README.md:12]: I think that would make sense.

**Review by vickenty — APPROVED** [review]

File: pkg/collector/README.md (hunk 3 above)
Comment (pgimalac) [reply pkg/collector/README.md]: Ok, done !

File: docs/proposals/README.md [hunk 4]
```diff
@@ -1,6 +1,3 @@
 # Design proposals
 
 This directory contains accepted design proposals.
```
Comment (GustavoCaso) [inline docs/proposals/README.md:3]: Maybe for a separate PR, but we barely use the proposals folder. We might as well move the proposals to the https://github.com/DataDog/architecture repo

**Review by GustavoCaso — APPROVED** [review]

File: docs/proposals/README.md (hunk 4 above)
Comment (pgimalac) [reply docs/proposals/README.md:3]: Why not, asking the Platform team (co-owning the folder with docs) about it

Comment (GustavoCaso) [reply docs/proposals/README.md:3]: Sure 😄 
@DataDog/documentation @DataDog/agent-platform, any thought about keeping this here or moving it to the https://github.com/DataDog/architecture repo?

Comment (hestonhoffman) [reply docs/proposals/README.md:3]: No issue with moving it from the docs side

**Review by jhgilbert — APPROVED** [review]

**Review by sblumenthal — APPROVED** [review]

File: docs/cloud-workload-security/backend.md [hunk 5]
```diff
@@ -3369,5 +3369,5 @@ CSM Threats logs have the following JSON schema:
 
 
 
-[1]: /security/threats/
-[2]: /security/threats/agent_expressions
+[1]: https://docs.datadoghq.com/security/threats/
```
Comment (paulcacheux) [inline docs/cloud-workload-security/backend.md:3372]: Those files are directly imported by the docs website, so the relative path actually works there. We would like to keep it as is

**Review by paulcacheux — CHANGES_REQUESTED** [review]

File: docs/cloud-workload-security/backend.md (hunk 5 above)
Comment (pgimalac) [reply docs/cloud-workload-security/backend.md:3372]: Oh ok I had no idea my bad, I'll revert that
Unfortunately it means those links are not working when looking from the Github UI...

**Review by paulcacheux — APPROVED** [review]

File: pkg/collector/README.md [hunk 6]
```diff
@@ -7,7 +7,4 @@ a check are all operations implemented in this package or one of the subpackages
 For further details, please refer to the specific READMEs:
 
 * [check](check/README.md)
-* [corechecks](corechecks/README.md)
-* [py](py/README.md)
-* [runner](runner/README.md)
```
Comment (chouetz) [inline pkg/collector/README.md:12]: you don't have alternatives for these missing ones I guess

**Review by chouetz — APPROVED** [review]

File: pkg/collector/README.md (hunk 6 above)
Comment (pgimalac) [reply pkg/collector/README.md:12]: Indeed there is just no README in those three directories...
I initially just set a link to the directory instead, but when I asked if they should be removed the answer was positive https://github.com/DataDog/datadog-agent/pull/21604#discussion_r1429810106

**Pull request merged**